inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -409,54 +404,30 @@ void MKLDNNROIPoolingNode::initSupportedPrimitiveDescriptors() {
 }
 
 void MKLDNNROIPoolingNode::createPrimitive() {
-    auto selectedPrimitiveDescriptor = getSelectedPrimitiveDescriptor();
-    if (!selectedPrimitiveDescriptor)
+    auto selectedPD = getSelectedPrimitiveDescriptor();
+    if (!selectedPD)
         IE_THROW() << "CPU ROI Pooling node with name '" << getName() << "' doesn't have primitive descriptors.";
-    auto config = selectedPrimitiveDescriptor->getConfig();
-
-    const int simd_w = mayiuse(cpu::x64::avx512_common) ? 16 : 8;
-    jpp.c_block = simd_w;
-
-    auto inDims = getParentEdgeAt(0)->getMemory().getStaticDims();
-    auto outDims = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
-
-    jpp.mb = outDims[0];
-    jpp.c = rnd_up(inDims[1], simd_w);
-    jpp.ih = inDims[2];
-    jpp.iw = inDims[3];
-    jpp.oh = outDims[2];
-    jpp.ow = outDims[3];
 
-    jpp.spatial_scale = spatial_scale;
-    jpp.pooled_h = pooled_h;
-    jpp.pooled_w = pooled_w;
+    refParams.c_block = mayiuse(cpu::x64::avx512_common) ? 16 : 8;;
+    refParams.nb_c_blocking = mayiuse(cpu::x64::avx512_common) ? 15 : 7;
+    refParams.alg = getAlgorithm();
 
-    jpp.nb_c = jpp.c / jpp.c_block;
+    const auto& config = selectedPD->getConfig();
+    refParams.src_prc = config.inConfs[0].desc->getPrecision();
+    refParams.dst_prc = config.outConfs[0].desc->getPrecision();
+    refParams.src_data_size = refParams.src_prc.size();
+    refParams.dst_data_size = refParams.dst_prc.size();
 
-    jpp.nb_c_blocking = mayiuse(cpu::x64::avx512_common) ? 15 : 7;
-
-    auto selectedPD = getSelectedPrimitiveDescriptor();
-    jpp.src_prc = selectedPD->getConfig().inConfs[0].desc->getPrecision();
-    jpp.dst_prc = selectedPD->getConfig().outConfs[0].desc->getPrecision();
-    jpp.src_data_size = jpp.src_prc.size();
-    jpp.dst_data_size = jpp.dst_prc.size();
-
-    jpp.alg = getAlgorithm();
-
-    if (mayiuse(cpu::x64::avx512_common)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::avx512_common>(jpp));
-    } else if (mayiuse(cpu::x64::avx2)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::avx2>(jpp));
-    } else if (mayiuse(cpu::x64::sse41)) {
-        roi_pooling_kernel.reset(new jit_uni_roi_pooling_kernel_f32<cpu::x64::sse41>(jpp));
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
     }
-
-    if (roi_pooling_kernel)
-        roi_pooling_kernel->create_ker();
 }
 
 template<typename T>
 void MKLDNNROIPoolingNode::execute() {
+    const auto& jpp = execPtr == nullptr ? refParams : execPtr->getJpp();
```

Comment:
Why refParams couldn't be used in both cases?

Reply:
Corrected, thank you!